cw/capture.py

```diff
@@ -991,6 +992,182 @@ def kmac_fvsr_key_batch(ctx: typer.Context,
     capture_end(ctx.obj.cfg)
 
 
+def capture_ecdsa_secret_keygen(ot, ktp, fw_bin, pll_frequency, capture_cfg):
+    """Capture traces for ECDSA P256/P384 secret key generation.
+
+    Uses a fixed seed and generates several random masks. For the corresponding
+    driver, check:
+    <opentitan_repo_root>/sw/device/sca/ecc256_keygen_serial.c
+
+    Args:
+      ot: Initialized OpenTitan target.
+      ktp: Key and plaintext generator.
+      fw_bin: Firmware binary.
+      pll_frequency: Output frequency of the FPGA PLL.
+      capture_cfg: Capture configuration from the yaml file
+    """
+    # Currently, we need to reprogram the firmware before every sign operation
+    # TODO: Investigate if this is needed.
+    reset_firmware = False
+
+    # OTBN operations are long. CW-Husky can store only 131070 samples
+    # in the non-stream mode. In case we want to collect more samples,
+    # we can run the operation multiple times with different adc_offset
+    # values.
+    # The trace from each run is kept in a section, and all sections are
+    # concatenated to create the final trace.
+    fifo_size = 131070
+    num_sections = capture_cfg["num_samples"] // fifo_size
+    print(f"num_sections = {num_sections}")
```

Comment:
I think this shouldn't be needed. Maybe it's best to start from the capture routine used for SHA3/KMAC for this. The code snippets should be small. If if they are 500 instructions and we oversample by 20x, the individual trace will just be 10000 samples.